### PR TITLE
Remove built-in notification system

### PR DESCRIPTION
**Pull request by CKolkey**
This comes from before plugins like https://github.com/rcarriga/nvim-notify were popular. Instead of re-implementing their work, we can just use `vim.notify()`.

@ten3roberts Any objection to dropping the home-made notifications?

### CODE REVIEW

**Review by ten3roberts — COMMENTED** [review]
Looks very thorough. I am very happy that all the existing "branches" of logs also have been updated and simplified.

My only question regarding this is how it behaves when the user does not use `nvim-notify` or `noice`. From what I recall, it will print in the command area, and testing now causes the dreaded `Press ENTER or type command to continue` if you do `:lua vim.notify("Foo") vim.notify("Bar")`

Would that be an acceptable behavior of blocking neovim and input with that?

Not sure how we would get around that

File: lua/neogit.lua [hunk 1]
```diff
@@ -33,7 +33,7 @@ function M.setup(opts)
   M.cli = M.lib.git.cli
   M.popups = require("neogit.popups")
   M.config = config
-  M.notif = require("neogit.lib.notification")
+  M.notification = require("neogit.lib.notification")
```
[the remark this answers is not in the document]
Comment (CKolkey) [reply lua/neogit.lua:36]: Hahaha, exactly. Glad you approve ;)

**Comment by CKolkey** [comment]
> My only question regarding this is how it behaves when the user does not use `nvim-notify` or `noice`. From what I recall, it will print in the command area, and testing now causes the dreaded `Press ENTER or type command to continue` if you do `:lua vim.notify("Foo") vim.notify("Bar")`

So, agreed, that would suck. But... that doesn't seem to happen? I disabled `nvim-notify` in my config, and the messages indeed do now print to the command area, but I'm not faced with a "hit-enter" prompt :shrug: The messages just print nicely. Maybe it's only an issue if the `vim.notify()` calls are _very_ close together, which we're not really doing.

**Comment by ten3roberts** [comment]
Hmm, it happens for me if I do `vim.notify()` twics in the same `:lua` command

From what I've read in the docs it happens if there are more than one messages in one *redraw*

**Comment by CKolkey** [comment]
Ah, ok, I think that won't be an issue then.

**Comment by CKolkey** [comment]
Oops, yeah, good looking out 😅

**Pull request merged**